### PR TITLE
feat: json path utils

### PR DESCRIPTION
**Pull request by anton62k**
## Summary by CodeRabbit

* **New Features**
  * Added JSON path utilities to parse path strings and to read, set, and check values within nested data structures.
  * Added robust deep-equality comparison for primitives, arrays, and plain objects, handling null/undefined and mixed nested structures.

* **Tests**
  * Added comprehensive unit tests covering path parsing, retrieval, mutation, existence checks, deep equality, and numerous edge cases.

### CODE REVIEW

File: src/lib/json-path-utils.ts [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+ * deepEqual(undefined, undefined)            // true
+ */
+export function deepEqual(a: unknown, b: unknown): boolean {
+  if (a === b) {
+    return true;
+  }
+
+  if (a == null && b == null) {
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (anton62k) [reply src/lib/json-path-utils.ts]: done
